demo/rails/app/views/captcha/index.html.erb

```diff
@@ -36,11 +36,21 @@
     <%= submit_tag %>
   <% end %>
   <script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit" async defer></script>
-  <%= link_to 'Single ?', '?' %>
+<% elsif params[:invisible] %>
+  <script>
+    function onSubmitCallback() {
```

Comment:
Not sure what you mean with being CSP compliant. Isn't there more inline JavaScript on this file already? And would making an anonymous function and storing it in a `var` be enough to deal with the generic name issue?